src/main.rs

```diff
@@ -1,120 +1,116 @@
 use chrono::{NaiveDate, NaiveDateTime};
+use clap::Parser;
 use comrak::{markdown_to_html, ComrakOptions};
 use frontmatter_gen::{extract, Frontmatter, Value};
+use fs_extra::dir::{copy, CopyOptions};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 use std::fs;
 use std::io;
-use std::path::{Path, PathBuf};
+use std::path::Path;
 use std::process;
+use std::sync::Arc;
 use tera::{Context, Tera};
 use walkdir::WalkDir;
 
+mod cli; // Import the CLI module
+mod server; // Import the server module
+
 fn main() -> io::Result<()> {
-    // Argument Parsing
-    let args: Vec<String> = std::env::args().collect();
-    if args.len() < 2 {
-        eprintln!("Usage: program <folder>");
-        process::exit(1);
-    }
-    let folder = PathBuf::from(&args[1]);
+    let args = cli::Cli::parse();
+
+    let input_folder = args.input_folder;
+    let output_folder = Arc::new(args.output_folder); // Convertemos para Arc<PathBuf>
+    let serve = args.serve;
+    let config_path = input_folder.join(args.config);
 
     // Initialize site data
-    let marmite = match fs::read_to_string("marmite.yaml") {
-        Ok(data) => data,
-        Err(e) => {
-            eprintln!("Unable to read marmite.yaml: {}", e);
-            process::exit(1);
-        }
-    };
+    let marmite = fs::read_to_string(config_path).unwrap_or_else(|e| {
+        eprintln!("Unable to read config file: {}", e);
+        process::exit(1);
+    });
+
```

Comment:
An important point here was the switch to `Arc<PathBuf>`, which improves the management of the output path when it needs to be shared between different parts of the code, especially in environments involving multiple threads (I was unsure how much it could demand when processing/rendering many *.md and *.js files, along with other formats). I'm also using `unwrap_or_else` instead of `unwrap`, which seems to improve handling unexpected crashes, as now errors are treated in a more controlled way with clear messages.